examples/wasm_custom_event.rs

```diff
@@ -0,0 +1,130 @@
+/// This example will show how to call a Rust function from JavaScript
+/// You should see the number 42 appear in browser console output when you click the button
+
+#[cfg(not(wasm_platform))]
+pub fn main() {
+    panic!("This example is only meant to be compiled for wasm target")
+}
+
+#[cfg(wasm_platform)]
+pub fn main() {
+    panic!("Please run `cargo run-wasm --example wasm_custom_event`")
+}
+
+#[cfg(wasm_platform)]
+mod wasm {
+    use std::cell::RefCell;
+
+    use wasm_bindgen::prelude::*;
+    use wasm_bindgen::JsCast;
+    use web_sys::HtmlScriptElement;
+    use winit::event::Event;
+    use winit::event::WindowEvent;
+    use winit::event_loop::EventLoop;
+    use winit::event_loop::EventLoopBuilder;
+    use winit::event_loop::EventLoopProxy;
+    use winit::window::Window;
+    use winit::window::WindowBuilder;
+
+    // Because EventLoopProxy is not Send, we need to wrap it in a RefCell and use thread_local!
+    thread_local! {
+        pub static EVENT_LOOP_PROXY: RefCell<Option<EventLoopProxy<CustomEvent>>> = RefCell::new(None);
+    }
+
+    // Function to be called from JS
+    fn wasm_call() -> u32 {
+        42
+    }
+
+    #[derive(Debug, Clone, Copy)]
+    pub enum CustomEvent {
+        WasmCall,
+    }
```

Comment:
Let's make a slightly more useful example. Instead of just printing 42 we could e.g. use `Window::set_inner_size()`, which is a valid use-case and would also show how to pass parameters. This would require two input fields for width and height and a button to send.

Reply:
Awaiting if the cargo-wasm-runner will go through first. https://github.com/rukai/cargo-run-wasm/pull/36